test/e2e-test-application/src/app/project/project.component.ts

```diff
@@ -129,14 +130,21 @@ export class ProjectComponent implements OnInit, OnDestroy {
   }
 
   openAsModal() {
-    linkManager()
-      .openAsModal('/projects/pr2/settings', {
-        title: 'microfrontend in a modal',
-        size: 'm'
-      })
-      .then(() => {
-        document.getElementById('promiseTest').innerHTML = 'promise resolved!';
-      });
+    const modalPath = '/projects/pr2/settings';
+    const modalSettings: ModalSettings = {
+      size: 'm',
+      title: 'microfrontend in a modal'
+    };
+    const resolvePromise = () => {
+      document.getElementById('promiseTest').innerHTML = 'promise resolved!';
+    };
+
+    if (Object.prototype.toString.call(linkManager().openAsModal) === '[object Promise]') {
+      ((linkManager().openAsModal(modalPath, modalSettings) as unknown) as Promise<void>).then(() => resolvePromise());
+    } else {
+      linkManager().openAsModal(modalPath, modalSettings);
+      setTimeout(() => resolvePromise(), 500);
```

Comment:
Why the change here. The **e2e-test-application** may be often used as a showcase of our features to Developers. This change here suggests it cant be done otherwise. As a Dev I wouldn't expect to have to do all these checks to open a modal. Question is why was this necessary here, and can it be reverted

Reply:
Looks like leftover after testing - now original version works as expected